pmd-apex/src/main/java/net/sourceforge/pmd/lang/apex/ast/ApexParser.java

```diff
@@ -4,23 +4,41 @@
 
 package net.sourceforge.pmd.lang.apex.ast;
 
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
 import net.sourceforge.pmd.annotation.InternalApi;
 import net.sourceforge.pmd.lang.apex.ApexLanguageProcessor;
 import net.sourceforge.pmd.lang.ast.ParseException;
 import net.sourceforge.pmd.lang.ast.Parser;
 
 import com.google.summit.SummitAST;
+import com.google.summit.translation.Translate;
 import com.google.summit.ast.CompilationUnit;
 
 @InternalApi
 public final class ApexParser implements Parser {
 
     public ApexParser() {
+        Logger.getLogger(Translate.class.getName()).setLevel(Level.WARNING);
     }
 
     @Override
     public ASTApexFile parse(final ParserTask task) {
-        final CompilationUnit astRoot = SummitAST.INSTANCE.parseAndTranslate(task.getTextDocument().getText().toString(), null);
+        CompilationUnit astRoot = null;
+
+        PrintStream err = System.err;
+        try {
+            // Redirect System.err to suppress ANTLR warning about runtime/compilation version mismatch.
+            // See: org.antlr.v4.runtime.RuntimeMetadata
+            System.setErr(new PrintStream(new ByteArrayOutputStream()));
```

Comment:
Interesting...

Note: we get this already in master because of ApexLink using a different Antlr version...
Note2: See also ApexCommentBuilder

Reply:
I've created a separate helper class AntlrVersionCheckSuppression for now.